### DISCUSSION TITLE
How to detect changes happening in the workspace?

### DISCUSSION THREAD

**Discussion by YathindraNikshipth** [eclipse-theia/theia#14360] [General] [answered]
I want to monitor changes in the open workspace, such as file creation, deletion, saving, and renaming. Is there an existing service that provides these events?

**Comment by msujew** [comment, marked as answer]
Hey @YathindraNikshipth,

you can use the `FileService` service to register the appropriate events. Note that there are two different kinds of workspace changes; Those who are happening on the file system (from the outside) and those initiated by the user. Here's an example on how to hook into both:

https://github.com/eclipse-theia/theia/blob/bcf96a394e9cb082dbcb93060eec2e380e704504/packages/collaboration/src/browser/collaboration-instance.ts#L472-L517

**Reply by YathindraNikshipth** [reply]
@msujew It's helpful, Thank you

**Reply by YathindraNikshipth** [reply]
hey @msujew  I would like to clarify whether the onDidRunOperation event only triggers for file-related changes. Additionally, could you let me know how to fetch changes related to folders? Thank you for your help.

**Reply by msujew** [reply]
`onDidRunOperation` triggers for all file system operations triggered from the UI of the application. As far as I know, both events also trigger for folder changes, not only files.

**Reply by YathindraNikshipth** [reply]
Hi @msujew, I was looking into the FileWatcherService to specifically monitor changes in certain folders, but I'm unsure how to configure it for that purpose. Could you please provide some guidance?
Thank you!
